### DISCUSSION TITLE
How to build RDKit documentation locally

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#5801] [Q&A] [answered]
I'd like to start contributing to the RDKit documentation. Are there instructions for how to build the docs locally? The only [section I could find about contributing](https://www.rdkit.org/docs/Cookbook.html#feedback-and-contributing) doesn't include detailed instructions.

So far I've:

1. Forked and cloned the repo
2. In a conda environment, installed recommonmark using `conda install recommonmark`
3. Tried running `sphinx-build` using the Makefile in `Docs/Book`, and am getting this error:

```
(my-rdkit-env) myusername@MacBook-Air rdkit % sphinx-build -M Docs/Book/Makefile Docs/Book _build
Running Sphinx v5.0.2

Sphinx error:
Builder name Docs/Book/Makefile not registered or available through entry point
```

**Comment by greglandrum** [comment, marked as answer]
@bertiewooster thanks for the interest in contributing to the docs!

To build a local copy you need to run the command: `make html` from the $RDBASE/Docs/Book directory
Since this also tries to copy in the C++ API documentation and you don't have a local C++ build of the RDKit, this will fail for you with an error message like:
```
cp: cannot stat '/Code/docs/*': No such file or directory
```
Unless you're interested in working on the C++ docs, you can work around this by editing `$RDBASE/Docs/Book/Makefile` and changing line 55 from:
```
html: $(RSTS_FROM_MD) apidocs cppexamples
```
to
```
html: $(RSTS_FROM_MD) cppexamples # apidocs 
```
Once you've done that running `make html` should work.

**Reply by greglandrum** [reply]
I can make this easier and will go ahead and do so.

**Reply by bertiewooster** [reply]
@greglandrum Those instructions worked for me, once I made that change in the Makefile--I'm now able to build the docs locally.

Great, making it easier for people to contribute to the docs can spur more people to do so. Once you've made that change, I'd be happy to add to the docs a section on how to contribute, including your detailed instructions on how to build locally. It would be good to put that in a place where people can easily find it, for example in [The RDKit Documentation](https://www.rdkit.org/docs/index.html), perhaps within [An overview of the RDKit](https://www.rdkit.org/docs/Overview.html)--a new section just above [The Contrib Directory](https://www.rdkit.org/docs/Overview.html#the-contrib-directory) seems like a logical place.

**Reply by greglandrum** [reply]
@bertiewooster sounds great. Thanks.

**Reply by bertiewooster** [reply]
> I can make this easier and will go ahead and do so.

In pull request #5804.

**Comment by vfscalfani** [comment]
Hi @bertiewooster. You may find my poster on the Cookbook helpful too. I have been meaning to add new recipes. Let me know if you need any help.

https://github.com/rdkit/UGM_2020/blob/master/Posters/Scalfani_VF_RDKitCookbook_2020_poster.pdf

Vin

**Reply by bertiewooster** [reply]
@vfscalfani thanks, good to know that your poster has the `make doctest` and `make html` commands.
